src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine_MapCreation.cs

```diff
@@ -19,48 +22,56 @@ namespace Microsoft.CodeAnalysis.FindSymbols
 
     internal partial class FindReferencesSearchEngine
     {
+        private static readonly Func<Project, DocumentMap> s_createDocumentMap = _ => new DocumentMap();
+
         private async Task<ProjectToDocumentMap> CreateProjectToDocumentMapAsync(ProjectMap projectMap)
         {
             using (Logger.LogBlock(FunctionId.FindReference_CreateDocumentMapAsync, _cancellationToken))
             {
-                var finalMap = new ProjectToDocumentMap();
+                using var _ = ArrayBuilder<Task<(ImmutableArray<Document>, ISymbol, IReferenceFinder)>>.GetInstance(out var tasks);
```

Comment:
💡 In this case, we know the exact number of tasks. We should create the required array directly.

Reply:
that's reasonable.  i can make that cahnge in one of my other FAR PRs that build off of this. :)